KubeArmor/enforcer/appArmorProfile.go

```diff
@@ -23,8 +23,8 @@ func (ae *AppArmorEnforcer) ResolvedProcessWhiteListConflicts(processWhiteList *
 	copy(prunedProcessWhiteList, *processWhiteList)
 	numOfRemovedElements := 0
 
-	for index, line := range *processWhiteList {
-		for source := range fromSources {
+	for source := range fromSources {
```

Comment:
I am not sure how this fixes segfault issue! ... is `processWhiteList` nil when there are no fromSources? Such that if we swap the for conditions, the fromSources are checked first and the *processWhiteList check is never hit?

Reply:
@nyrahul  `processWhiteList` is not nil. The issue is that in source, the second time, the index is not incremented. That index should increment in second loop. 

`prunedProcessWhiteList = kl.RemoveStringElement(prunedProcessWhiteList, index-numOfRemovedElements)`

In this case, in second loop, the second argument is 0-1, which is -1 for a string slice. This causes array index out of bounds.

Reply:
I don't still get the point. can you explain the problem a bit more?